grouper/fe/handlers/group_edit_member.py

```diff
@@ -156,3 +134,22 @@ def post(self, *args: Any, **kwargs: Any) -> None:
             )
 
         return self.redirect("/groups/{}?refresh=yes".format(group.name))
+
+    def _get_form(self, my_role: str, member_type: str) -> GroupEditMemberForm:
+        """Get the form with possible role options filled in.
+
+        Owners can edit themselves to downgrade their membership, but not upgrade their
+        membership.  So owners can make themselves members, np-owners, or managers, but np-owners
+        cannot make themselves owners.  The role of a group cannot be changed from member.
+        """
+        form = GroupEditMemberForm(self.request.arguments)
+        form.role.choices = [["member", "Member"]]
+        if my_role in ("owner", "np-owner") and member_type != "group":
+            form.role.choices.append(["manager", "Manager"])
```

Comment:
Isn't np-owner -> manager technically not a downgrade, as it'd give them permissions? Unless our help text is wrong about `a manager is a member who is additionally allowed to process requests to join a group`

Reply:
Yes. I had completely forgotten about that. Thank you!